mdata/store_cassandra.go

```diff
@@ -77,9 +78,37 @@ type cassandraStore struct {
 	session     *gocql.Session
 	writeQueues []chan *ChunkWriteRequest
 	readQueue   chan *ChunkReadRequest
+	ttlTables   map[uint32]string
 }
 
-func NewCassandraStore(addrs, keyspace, consistency, CaPath, Username, Password, hostSelectionPolicy string, timeout, readers, writers, readqsize, writeqsize, retries, protoVer int, ssl, auth, hostVerification bool) (*cassandraStore, error) {
+func (c *cassandraStore) ensureTtlTables(ttls []uint32, keyspace string) error {
+	for _, ttl := range ttls {
+		// calculate table name and window
+		var window int = 1
+		for tmp := ttl / (24 * 60 * 60); tmp > 1; tmp = tmp / 2 {
+			window *= 2
+		}
+		table_name := fmt.Sprintf("metric_%d", window)
+
+		// table isn't know to exist, ensure it exists
+		err := c.session.Query(fmt.Sprintf(table_schema, keyspace, table_name, window)).Exec()
```

Comment:
Actually we can only how some how large the time range of most queries is, it doesn't necessarily need to be tied to the `TTL`. So I was also considering to add some option like `compaction-window-factor` and the I'd just:

```
var window int = 1
    for tmp := ttl / (24 * 60 * 60); tmp > 1; tmp = tmp / 2 {
    window *= 2
}
window = window / compactionWindowFactor
```

Because in our case one ratio between `TTL` and `compaction window` might make sense, but if somebody is mostly interested in archiving until forever but they always just query the last day then our assumption might be sub optimal for them.